From: Andreas Hartmetz <ahartmetz@gmail.com>
Date: Sat, 12 Nov 2016 18:46:20 +0100
Subject: Also change the resize mode the other way...

---
 src/filewidgets/kdiroperatordetailview.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/filewidgets/kdiroperatordetailview.cpp b/src/filewidgets/kdiroperatordetailview.cpp
index 491aea57..32e82183 100644
--- a/src/filewidgets/kdiroperatordetailview.cpp
+++ b/src/filewidgets/kdiroperatordetailview.cpp
@@ -91,6 +91,8 @@ bool KDirOperatorDetailView::setViewMode(KFile::FileView viewMode)
     // using KDirOperator in horizontally limited parts of an app.
     if (tree && m_hideDetailColumns) {
         header()->setSectionResizeMode(QHeaderView::ResizeToContents);
+    } else {
+        header()->setSectionResizeMode(QHeaderView::Interactive);
     }
 
     return true;
