Index: kio/src/ioslaves/help/kio_help.cpp
===================================================================
--- kio.orig/src/ioslaves/help/kio_help.cpp	2015-11-19 14:33:04.080944405 +0100
+++ kio/src/ioslaves/help/kio_help.cpp	2015-11-19 14:36:35.272354243 +0100
@@ -49,7 +49,8 @@
     QStringList search;
 
     // assemble the local search paths
-    const QStringList localDoc = QStandardPaths::locateAll(QStandardPaths::GenericDataLocation, QStringLiteral("doc/HTML"), QStandardPaths::LocateDirectory);
+    QStringList localDoc = QStandardPaths::locateAll(QStandardPaths::GenericDataLocation, QStringLiteral("doc/HTML"), QStandardPaths::LocateDirectory);
+    localDoc << QStandardPaths::locateAll(QStandardPaths::GenericDataLocation,QStringLiteral("doc/kde/HTML"), QStandardPaths::LocateDirectory);
 
     QStringList langs = QLocale().uiLanguages();
     langs.append(QStringLiteral("en"));
